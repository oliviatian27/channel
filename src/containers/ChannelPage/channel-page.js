import React,{Component} from 'react';

import loadFromServer from '../../utils/adapter'
import ChannelList from '../../components/channel-list';
// import './style.scss';


class  ChannelPage extends Component{
  constructor(props) {
     super(props);
     this.state = {
       channelList: []
     };
   }

  componentDidMount(){
     loadFromServer(channelList=>{
       this.setState({channelList})
     })
   }
  render(){
    let {channelList}=this.state

    if(channelList.length===0){
      return <span className="loading">No Channel yet </span>
    }
    return (
        <div className="container">
         {channelList.map(list=><ChannelList key={list.date} list={list}/>)}
        </div>
    );
  }

}



export default ChannelPage;
