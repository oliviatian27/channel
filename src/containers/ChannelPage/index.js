export { default } from './channel-page';
