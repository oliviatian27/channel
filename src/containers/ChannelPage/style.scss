.app-wrapper {
  max-width: calc(768px + 16px * 2);
  margin: 0 auto;
  display: grid;
  min-height: 100%;
  padding: 0 16px;
  grid-template-columns: repeat(14, 1fr);
}
