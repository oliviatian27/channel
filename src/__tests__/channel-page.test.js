import React from 'react';
import { shallow,mount,render } from 'enzyme';

import ChannelPage from '../containers/ChannelPage';
import loadFromServer from '../utils/adapter'

describe('ChannelPage tests', () => {

  it('current state render', () => {
  //   beforeEach(() => {
  //   fetch.resetMocks()
  // })
    const fakeData = [{"title":"1","time": "2018-01-03 22:00:00"},{"title":"2","time": "2018-01-04 10:00:00"},
    {"title":"3","time": "2018-01-03 12:00:00"},{"title":"4","time": "2018-01-04 11:00:00"}]

    // const flushPromises=()=>( new Promise(resolve => setImmediate(resolve)))

    fetch.mockResponse(JSON.stringify(fakeData), {status: 200});


    const wrapper = mount(<ChannelPage />)


    // componentDidMount has been run, but the then()-function of
    // the promise (returned from fetch) has not been executed
    //  => callback not invoked
    //  => setState not invoked
    //  => no rendering of channel page

    expect(wrapper.find('.loading')).toHaveLength(1);
    expect(wrapper.find('.container')).toHaveLength(0);

    loadFromServer((json) => {
        // then()-function has now been executed
        //  => success-callback is called
        //  => setState is called
        //  => new_wrapper is re-rendered with fakeData we passed to our mock
        wrapper.setState({channelList:json})
        // const new_wrapper = wrapper.render()
        //
        expect(wrapper.find('Channel').length).toEqual(fakeData.length);
        expect(wrapper.find('.loading')).toHaveLength(0);
        //
        // // just to make sure: fetch should have been called exactly once
        // expect(fetch.mock.calls).toHaveLength(1);

        //expect to render two ChannelList components 2018-01-03 and 2018-01-04
        expect(wrapper.find('ChannelList').length).toEqual(2);
        expect(wrapper.find('ChannelDate').at(0).prop("date")).toEqual("2018-01-03");
        expect(wrapper.find('ChannelDate').at(1).prop("date")).toEqual("2018-01-04");

        wrapper.unmount()
    });

  });

  it('renders ChannelPage', () => {
    const wrapper = shallow(<ChannelPage />)

    // Expect the wrapper object to be defined
    expect(wrapper.find('.container')).toBeDefined();
    expect(wrapper.find('.container')).toMatchSnapshot();
  });



})
