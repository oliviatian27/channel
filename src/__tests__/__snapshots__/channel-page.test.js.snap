// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChannelPage tests renders ChannelPage 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): ShallowWrapper {
    Symbol(enzyme.__root__): [Circular],
    Symbol(enzyme.__unrendered__): <ChannelPage />,
    Symbol(enzyme.__renderer__): Object {
      "batchedUpdates": [Function],
      "getNode": [Function],
      "render": [Function],
      "simulateError": [Function],
      "simulateEvent": [Function],
      "unmount": [Function],
    },
    Symbol(enzyme.__node__): Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "No Channel yet ",
        "className": "loading",
      },
      "ref": null,
      "rendered": "No Channel yet ",
      "type": "span",
    },
    Symbol(enzyme.__nodes__): Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "No Channel yet ",
          "className": "loading",
        },
        "ref": null,
        "rendered": "No Channel yet ",
        "type": "span",
      },
    ],
    Symbol(enzyme.__options__): Object {
      "adapter": ReactSixteenAdapter {
        "options": Object {
          "enableComponentDidUpdateOnSetState": true,
          "lifecycles": Object {
            "componentDidUpdate": Object {
              "onSetState": true,
            },
            "getDerivedStateFromProps": true,
            "getSnapshotBeforeUpdate": true,
            "setState": Object {
              "skipsComponentDidUpdateOnNullish": true,
            },
          },
        },
      },
    },
  },
  Symbol(enzyme.__unrendered__): null,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): undefined,
  Symbol(enzyme.__nodes__): Array [],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__rootNodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "No Channel yet ",
        "className": "loading",
      },
      "ref": null,
      "rendered": "No Channel yet ",
      "type": "span",
    },
  ],
}
`;
